src/crimson/net/ProtocolV1.cc

```diff
@@ -25,6 +25,26 @@
 WRITE_RAW_ENCODER(ceph_msg_connect);
 WRITE_RAW_ENCODER(ceph_msg_connect_reply);
 
+namespace {
+template<typename T, typename=void>
+struct unaligned_t;
+
+template<typename T>
+struct unaligned_t<T, enable_if_t<std::is_pod_v<T>>>
+{
+  T value;
+} __attribute__((packed));
+
+template<typename T>
+inline T unaligned_load(const char* p)
+{
+  using unaligned = unaligned_t<T>;
```

Comment:
can simplify out the partial specialization of unaligned_t:
```c++
template <typename T>
inline T unaligned_load(const char* p)
{
  static_assert(std::is_pod_v<T>);
  struct unaligned { T value; } __attribute__((packed));
  ...
```

Reply:
@cbodley thanks for the suggestion! i have a better understanding of SFINAE now.